**Hypothesis Tests for One Population Mean when Sigma Is Known**

If the population standard deviation $(\sigma)$ is known, a hypothesis test performed for one population mean is called **one-mean $z$-test** or simply **$z$-test**.

A $z$-test is a hypothesis test for testing a population mean, $\mu$, against a supposed population mean $\mu_0$. The $z-$ test assumes normally distributed variables or a large sample size; then the **central limit theorem** guarantees a normally distributed **sampling distribution**. In addition, $\sigma$, the standard deviation of the population must be known. In real life applications this assumption is almost never fulfilled, and thus, the $z-$ test is rarely applied. However, it is the simplest hypothesis test and thus, a good subject to start with.

To perform the $z$-test we follow the step wise procedure, as shown in the table below. First, we showcase the **critical value approach**, then, in a second step, we repeat the analysis for the **$p$-value approach**.

\begin{array}{l}
\hline
\ \text{Step 1}  & \text{State the null hypothesis } H_0 \text{ and alternative hypothesis } H_A \text{.}\\
\ \text{Step 2}  & \text{Decide on the significance level, } \alpha\text{.} \\
\ \text{Step 3}  & \text{Compute the value of the test statistic.} \\
\ \text{Step 4a} & \text{Critical value approach: Determine the critical value.}
\\
\ \text{Step 4b} &\text{P-value approach: Determine the p-value.} \\
\ \text{Step 5a} & \text{Critical value approach: If the value of the test statistic falls in the rejection region, reject} H_0  \text{; otherwise, do not reject} H_0 \text{.}
\\
\ \text{Step 5b} & \text{P-value approach: If } p \le \alpha \text{, reject }H_0 \text{; otherwise, do not reject } H_0 \text{.} \\
\ \text{Step 6} &\text{Interpret the result of the hypothesis test.} \\
\hline 
\end{array}

**One-mean $z$-test: An example**

In this section we work with the students data set.

In [1]:
import pandas as pd

students_df = pd.read_csv("https://userpage.fu-berlin.de/soga/200/2010_data_sets/students.csv")

In order to showcase hypothesis testing we examine the average weight of students and compare it to the average weight of Europeans adults. [Walpole et al. (2012)](https://bmcpublichealth.biomedcentral.com/articles/10.1186/1471-2458-12-439) published data on the average body mass *(kg)* per region, including Europe. They report the mean the average body mass for the European adult population to be $70.8$ *kg*. We set $\mu_0$, the population mean accordinly, $\mu_0 = 70.8$. Unfortunately, owing to the methodological approach of Walpole et al. $(2012)$ they did not provide a standard deviation $(\sigma)$ of the weights of European adults. For demonstration purposes we assume that the weight data given in the *students* data set is a good approximation for the population of interest. Thus, we set $\sigma$ to the standard deviation of the weight variable in *students* data set.

In [2]:
mu_0 = 70.8
sigma = students_df['weight'].std()
sigma

8.635161688392094

Further, we take one random sample with a sample size of $n=14$. The sample consists of the weights in *kg* of $14$ randomly picked students from the *students* data set. Finally, we calculate the sample mean $(\bar{x})$, which is our sample statistic of interest. 

In [26]:
import numpy as np

n = 14

x_weight = np.random.choice(students_df['weight'], n)

x_bar = np.mean(x_weight)

x_bar

77.14285714285714

**Hypothesis testing: The critical value approach**

**Step 1: State the null hypothesis, $H_0$, and alternative hypothesis, $H_A$.**

The null hypothesis states that the average weight of students $(\mu)$ equals the average weight of European adults of $70.8$ *kg* $(\mu_0)$ as reported by Walpole et al. ($2012$). In other wordsm there is no difference in the mean weight of students and the mean weight of European adults.

$$H_0 : \mu = 70.8$$

For the purpose of illustration we test three alternating hypothesis.

**Alternative hypothesis 1**: The average weight of students does not equal the average weight of European adults. In other words there is a difference in the mean weight of students and the mean weight of European adults.

$$H_{A_1} : \mu \neq 70.8$$

**Alternative hypothesis 2**: The average weight of students is less than the average weight of European adults.

$$H_{A_2}: \mu < 70.8$$

**Alternative hypothesis 3**: The average weight of students is higher than the average weight of European adults.

$$H_{A_3} : \mu > 70.8$$

**Step 2: Decide on the significance level, $\alpha$.**

$$\alpha = 0.05$$

In [27]:
alpha = 0.05

**Step 3: Compute the value of the test statistic.**

The following equation is applied to calculate the test statistic $z$.

$$z = \frac{\bar{x}- \mu_0}{\sigma/\sqrt{n}}$$

In [29]:
import math

z = (x_bar - mu_0)/ (sigma / math.sqrt(n))

z

2.7483907236764704

**Step 4a: Determine the critical value.**


In order to  calculate the critical value we apply the <mark> ppf </mark> function in python. Recall, that we test for three alternative hypothesis $(H_{A_1}, H_{A_2}, \text{and } H_{A_3})$, thus, we have to calculate three critical values as well $(z_{A_1} = \pm z_{\alpha/2}, z_{A_2} = - z_{\alpha}, \text{and} z_{A_3} = +z_{\alpha})$.

In [35]:
from scipy.stats import norm

z_A1 = norm.ppf(1-alpha/2)
z_A2 = -norm.ppf(1-alpha)
z_A3 = norm.ppf(1-alpha)

print(z_A1,z_A2,z_A3)

1.959963984540054 -1.6448536269514722 1.6448536269514722


**Step 5a: If the value of the test statistic falls in the rejection region, reject $H_0$; otherwise, do not reject $H_0$.**

The value of the test statistic found in Step $3$ is $z \approx 2.75$. Recall, that we are investigating three alternative hypothesis ($H_{A_1}$, $H_{A_2}$ and $H_{A_3}$). Thus, we evaluate the rejection region for each particular hypothesis.

- **Alternative hypothesis $H_{A_1} : \mu \neq 70.8$**

Recall the critical values for $H_{A_1}$.

$$z_{A_1} = \pm z_{\alpha/2} = \pm 1.96$$

Does the test statistic ($z \approx 2.75$) fall in the rejection region? Yes, it does as $2.75 > 1.96$.

So we reject $H_0$. The test results are statistically significant at the $5$% level.

- **Alternative hypothesis $H_{A_1} : \mu < 70.8$**

Recall the critical value for $H_{A_2}$.

$$z_{A_2} = - z_{\alpha} = -1.64$$

Does the test statistic ($z \approx 2.75$) fall in the rejection region? No, it does not as $2.75 < -1.64$ is FALSE.

So we do not reject $H_0$. The test results are statistically significant at the $5$% level.

- **Alternative hypothesis $H_{A_3} : \mu > 70.8$**

Recall the critical value for $H_{A_3}$.

$$z_{A_3} = + z_{\alpha} = 1.64$$

Does the test statistic ($z \approx 2.75$) fall in the rejection region? Yes, it does as $2.75 > 1.64$.

So we reject $H_0$. The test results are statistically significant at the $5$% level.

**Step 6: Interpret the result of the hypothesis test.**

- **Alternative hypothesis $H_{A_1} : \mu \neq 70.8$**

At the $5$% significance level, the data provides sufficient evidence to conclude that the average weight of students differs from the average weight European adults.

- **Alternative hypothesis $H_{A_2} : \mu < 70.8$**

At the $5$% significance level, the data does not provide sufficient evidence to conclude that the average weight of students is less than the average weight of European adults.

- **Alternative hypothesis $H_{A_3} : \mu > 70.8$**

At the $5$% significance level, the data provides sufficient evidence to conclude that the average weight of students is higher than average weight of European adults.

**Hypothesis testing: The $p$-value approach**

**Step 1: State the null hypothesis, $H_0$, and alternative hypothesis, $H_A$.**

The null hypothesis states that the average weight of students $(\mu)$ equals the average weight of European adults of $70.8$ *kg* $(\mu_0)$ as reported by Walpole et al. ($2012$). In other wordsm there is no difference in the mean weight of students and the mean weight of European adults.

$$H_0 : \mu = 70.8$$

For the purpose of illustration we test three alternating hypothesis.

**Alternative hypothesis 1**: The average weight of students does not equal the average weight of European adults. In other words there is a difference in the mean weight of students and the mean weight of European adults.

$$H_{A_1} : \mu \neq 70.8$$

**Alternative hypothesis 2**: The average weight of students is less than the average weight of European adults.

$$H_{A_2}: \mu < 70.8$$

**Alternative hypothesis 3**: The average weight of students is higher than the average weight of European adults.

$$H_{A_3} : \mu > 70.8$$

**Step 2: Decide on the significance level, $\alpha$.**

$$\alpha = 0.05$$

In [37]:
alpha = 0.05

**Step 3: Compute the value of the test statistic.**

The following equation is applied to calculate the test statistic $z$.

$$z = \frac{\bar{x}- \mu_0}{\sigma/\sqrt{n}}$$

In [38]:
import math

z = (x_bar - mu_0)/ (sigma / math.sqrt(n))

z

2.7483907236764704

**Step 4b: Determine the $p$-value.**

In order to calculate the $p-value$ we apply the <mark> cdf </mark> function in Python. Recall, that we test for three alternative hypothesis ($H_{A_1}$, $H_{A_2}$ and $H_{A_3}$), thus we can calculate three $p$-values as well $(P(z_{A_1}), P(z_{A_2}), \text{and } P(z_{A_3}))$.

- **Alternative hypothesis $$H_{A_1}:\quad \mu \neq 70.8$$**

In [39]:
from scipy.stats import norm

upper = 1-norm.cdf(z)
lower = norm.cdf(-z)
p_z1 = upper + lower 
p_z1

0.005988859425891677

From step $3$, the value of the test statistic is $z \approx 2.75$. The test is two-sided, so the $p-$value is the probability of observing a value $z$ of $2.75$ or greater in magnitude of a $z$-value of $-2.75$ or lower in magnitude. We have $p \approx 0.006$.

- **Alternative hypothesis $H_{A_2} : \mu < 70.8$**

In [41]:
p_z2 = norm.cdf(z)
p_z2

0.9970055702870542

From step $3$, the value of the test statistic is $z \approx 2.75$. The test is left-tailed, so the $p$-value is the probability of observing a value $z$ of $2.75$ or lower in magnitude. We have $p \approx 1$.

- **Alternative hypothesis $H_{A_3} : \mu > 70.8$**

In [42]:
p_z3 = 1- norm.cdf(z)
p_z3

0.002994429712945834

From step $3$, the value of the test statistic is $z \approx 2.75$. The test is right-tailed, so the $p$-value is the probability of observing a value $z$ of $2.75$ or greater in magnitude. We have $p \approx 0.003$.

**Step 5b: If $p \leq \alpha$, reject $H_0$; otherwise, do not reject $H_0$.**

The $p$-value of the test statistic found in step $4$ is compared to the user defined significance level $\alpha$ of $5$%. Recall that we are investigating three alternative hypothesis ($H_{A_1}$, $H_{A_2}$ and $H_{A_3}$). Thus, we make comparison for each particular hypothesis.

- **Alternative hypothesis $$H_{A_1}:\quad \mu \neq 70.8$$**

$$0.006 \leq 0.05$$

The $p$-value is less than the specified significance level of $0.05$ we reject $H_{0}$. The test results are statistically significant at the $5$% level and provide very strong evidence against the null hypothesis.

- **Alternative hypothesis $$H_{A_1}:\quad \mu < 70.8$$**

$$1 \leq 0.05$$ 

is FALSE. The $p$-value is greater than the specified significance level of $0.05$ we do not reject $H_0$. The test results are statistically significant at the $5$% level and provide not sufficient evidence against the null hypothesis. 

- **Alternative hypothesis $$H_{A_1}:\quad \mu > 70.8$$**

$$0.003 \leq 0.05$$

The $p$-value is less than the specified significance level of $0.05$ we reject $H_0$. The test results are statistically significant at the $5$% level and provide very strong evidence against the null hypothesis.

**Step 6: Interpret the result of the hypothesis test.**

- **Alternative hypothesis $H_{A_1}:\mu \neq70.8$**

$p \approx 0.006$. At the $5$% significance level, the data provides very strong evidence to conclude that the average weight of students differs from the average weight European adults.

- **Alternative hypothesis $H_{A_1}:\mu < 70.8$**

$p \approx 1$. At the $5$% significance level, the data does not provide sufficient evidence to conclude that the average weight of students is less than the average weight of European adults.

- **Alternative hypothesis $H_{A_1}:\mu > 70.8$**

$p \approx 0.003$. At the $5$% significance level, the data provides very strong evidence to conclude that the average weight of students is higher than average weight of European adults.